cluster-autoscaler/cloudprovider/gce/gce_url.go

```diff
@@ -64,24 +61,32 @@ func ParseInstanceUrlRef(url string) (GceRef, error) {
 }
 
 // GenerateInstanceUrl generates url for instance.
-func GenerateInstanceUrl(ref GceRef) string {
+func GenerateInstanceUrl(domainUrl string, ref GceRef) string {
+	if domainUrl == "" {
+		domainUrl = defaultDomainUrl
+	}
+	instanceUrlTemplate := domainUrl + projectsSubstring + "%s/zones/%s/instances/%s"
 	return fmt.Sprintf(instanceUrlTemplate, ref.Project, ref.Zone, ref.Name)
 }
 
 // GenerateMigUrl generates url for instance.
-func GenerateMigUrl(ref GceRef) string {
+func GenerateMigUrl(domainUrl string, ref GceRef) string {
+	if domainUrl == "" {
+		domainUrl = defaultDomainUrl
+	}
+	migUrlTemplate := domainUrl + projectsSubstring + "%s/zones/%s/instanceGroups/%s"
 	return fmt.Sprintf(migUrlTemplate, ref.Project, ref.Zone, ref.Name)
 }
 
 func parseGceUrl(url, expectedResource string) (project string, zone string, name string, err error) {
```

Comment:
nit: I feel like this whole function could be just a single regexp check. What do you think about rewriting this?

Reply:
I replaced the checks with regexp, however I still left the split logic. Let me know if you think we should change it further

Reply:
I think this is good enough for now - ideal would be replacing the split with regexp as well, but this is already much better.